pkg/suggestion/nasrl_service.py

```diff
@@ -1,22 +1,227 @@
+                lstm_num_layers=self.suggestion_config['lstm_num_layers'],
+                lstm_keep_prob=self.suggestion_config['lstm_keep_prob'],
+                lr_init=self.suggestion_config['init_learning_rate'],
+                lr_dec_start=self.suggestion_config['lr_decay_start'],
+                lr_dec_every=self.suggestion_config['lr_decay_every'],
+                lr_dec_rate=self.suggestion_config['lr_decay_rate'],
+                l2_reg=self.suggestion_config['l2_reg'],
+                entropy_weight=self.suggestion_config['entropy_weight'],
+                bl_dec=self.suggestion_config['baseline_decay'],
+                optim_algo=self.suggestion_config['optimizer'],
+                skip_target=self.suggestion_config['skip-target'],
+                skip_weight=self.suggestion_config['skip-weight'],
+                name="Ctrl_"+self.current_study_id)
+
+            self.controller.build_trainer()
+
+        print("Suggestion for StudyJob {} has been initialized.".format(request.study_id))
 
     def GetSuggestions(self, request, context):
+        if request.study_id != self.current_study_id:
+            self.reset_controller(request)
+            self.is_first_run = True
+
+        print("-" * 80 + "\nSuggestion Step {}\n".format(self.ctrl_step) + "-" * 80)
+
+        with self.tf_graph.as_default():
+
+            saver = tf.train.Saver()
+
+            controller_ops = {
+                  "train_step": self.controller.train_step,
+                  "loss": self.controller.loss,
+                  "train_op": self.controller.train_op,
+                  "lr": self.controller.lr,
+                  "grad_norm": self.controller.grad_norm,
+                  "optimizer": self.controller.optimizer,
+                  "baseline": self.controller.baseline,
+                  "entropy": self.controller.sample_entropy,
+                  "sample_arc": self.controller.sample_arc,
+                  "skip_rate": self.controller.skip_rate}
+
+            run_ops = [
+                controller_ops["loss"],
+                controller_ops["entropy"],
+                controller_ops["lr"],
+                controller_ops["grad_norm"],
+                controller_ops["baseline"],
+                controller_ops["skip_rate"],
+                controller_ops["train_op"]]
+
+            if self.is_first_run:
+                print("First time running suggestion. Random architecture will be given. \n")
+                with tf.Session() as sess:
+                    sess.run(tf.global_variables_initializer())
+                    arc = sess.run(controller_ops["sample_arc"])
+                    saver.save(sess, self.ctrl_cache_file)
+
+                self.is_first_run = False
+
+            else:
+                with tf.Session() as sess:
+                    saver.restore(sess, self.ctrl_cache_file)
+
+                    valid_acc = self.controller.reward
+                    result = self.GetEvaluationResult(request.study_id)
```

Comment:
We can define optimizationtype of a studyjob, which can be `maximize` and `minimize`. saying if objectivevaluename is `loss`, we hope its value change smaller and smaller(optimizationtype is minimize).

GetEvaluationResult should return last value of objectivevaluename metrics. and return of GetEvaluationResult is used to feed controller.reward (which hopes bigger and bigger).

So if a studyjob wants to get suggestion from suggestion-nasrl with `minimize` optimizationtype, GetEvaluationResult should return `-(lastOfobjectivevalue)`?